File: src/main/java/ru/umeta/libraryintegration/json/ModsParseResult.java
package ru.umeta.libraryintegration.json;

import gov.loc.mods.v3.ModsDefinition;

/**
 * Created by Kirill Kosolapov (https://github.com/c-tash) on 14/04/2015.
 */
public class ModsParseResult extends ParseResult {
    private ModsDefinition modsDefinition;

    public ModsParseResult(String title, String isbn, String author, Integer publishYear, ModsDefinition
            modsDefinition) {
        super(title, isbn, author, publishYear);
        this.modsDefinition = modsDefinition;
    }

    public ModsDefinition getModsDefinition() {
        return modsDefinition;
    }

    public void setModsDefinition(ModsDefinition modsDefinition) {
        this.modsDefinition = modsDefinition;
    }

    @Override
    public ParseResult clone() {
        return new ModsParseResult(title, isbn, author, publishYear, modsDefinition);
    }
}

File: src/main/kotlin/ru/umeta/libraryintegration/json/ModsParseResult.kt
package ru.umeta.libraryintegration.json

import gov.loc.mods.v3.ModsDefinition

/**
 * Created by Kirill Kosolapov (https://github.com/c-tash) on 14/04/2015.
 */
class ModsParseResult(title: String,
                      isbn: String,
                      author: String,
                      publishYear: Int?,
                      var modsDefinition: ModsDefinition?)
: ParseResult(title, isbn, author, publishYear) {

    override fun clone(): ParseResult {
        return ModsParseResult(title, isbn, author, publishYear, modsDefinition)
    }
}
